### PR TITLE
Implement voltage sensor

### PR DESCRIPTION
**Pull request by jordans6**
Uses old battery.cpp and battery.h files to implement battery as a sensor

### CODE REVIEW

File: avionics/envs/board/src/sensors/battery.cpp [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+#endif
+    int inputValue = Hal::analogRead(m_batterySensorPin);
+    // map it to the range the analog out:
+    // 3300mV is the highest voltage Teensy can read.
+    float teensyVoltage = sensor_range_map(inputValue, 0, 1023, 0, 3300);
+    // converts output value from mV to V and divides by voltage divider
+    // value to calculate battery input voltage.
+    float batteryVoltage = (teensyVoltage / m_divider) / 1000;
+    data_[0] = batteryVoltage; 
+}
```
Comment (jesseli2002) [inline avionics/envs/board/src/sensors/battery.cpp]: It's probably better practice to leave a newline at the end of each file - check your IDE to see if there's an option to automatically add a newline (and not render it); e.g. VS Code has Render Final Newline & Insert Final Newline options

**Review by jesseli2002 — CHANGES_REQUESTED** [review]
Looks like battery voltage sensor has the unique property that it can be implemented entirely through already-provided hardware abstractions - no need to put separate code under `envs/board` and `envs/x86`. I don't think there's any actual code changes - just moving around of some files: 
- Delete old battery.h and battery.cpp files
- Move board/src/sensors/battery.cpp to common/src/sensors/battery.cpp
- Move board/include/sensors/battery_sensor.h to common/include/sensors/battery.hpp

File: avionics/envs/board/src/sensors/battery.cpp (hunk 1 above)
Comment (jordans6) [reply avionics/envs/board/src/sensors/battery.cpp]: Adjusted that setting now - 7eeedd5

**Comment by jordans6** [comment]
> Looks like battery voltage sensor has the unique property that it can be implemented entirely through already-provided hardware abstractions - no need to put separate code under `envs/board` and `envs/x86`. I don't think there's any actual code changes - just moving around of some files:
> 
> * Delete old battery.h and battery.cpp files
> * Move board/src/sensors/battery.cpp to common/src/sensors/battery.cpp
> * Move board/include/sensors/battery_sensor.h to common/include/sensors/battery.hpp

Made those changes
7eeedd5

**Review by jesseli2002 — APPROVED** [review]
LGTM

**Pull request merged**